tests/forms_tests/widget_tests/test_widget.py

```diff
@@ -15,3 +15,12 @@ def test_value_omitted_from_data(self):
 
     def test_no_trailing_newline_in_attrs(self):
         self.check_html(Input(), 'name', 'value', strict=True, html='<input type="None" name="name" value="value" />')
+
+    def test_attrs_with_type_passed(self):
+        attrs = {'type': 'date'}
+        self.check_html(TextInput(attrs), 'name', 'value', strict=True, html=(
```

Comment:
TextInput -> Input? I suppose a `test_input.py` file would be better. I wasn't sure about the `test_no_trailing_newline_in_attrs` test -- it's meant to test a template rather than Python code -- probably I could have clarified that.
`strict=True` isn't needed since the newline isn't being tested.